.github/workflows/nightly.yaml

```diff
@@ -1,6 +1,9 @@
 on:
-  schedule:
-    - cron: '8 0 * * *'
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
```

Comment:
You should definitely keep `workflow_dispatch:` so we can just run it manually when we have problems.